### PR TITLE
Added error component to the dropdown pages, service and singleDirect…

### PR DESCRIPTION
**Pull request by RobCatton**
…ion. Also updated snapshots and a few tests.

# Description

-   Bringing the direction and service pages in line with others in regards to error handling.

# Testing instructions

-   Run the make command in the fdbt-dev repo and test the site.

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [x] Able to run pr locally
-   [x] Lighthouse performed
-   [x] Followed acceptance criteria
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have made corresponding changes to the documentation if applicable
-   [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/pages/service.tsx [hunk 1]
```diff
@@ -71,6 +86,7 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
     }
 
     const operatorInfo = JSON.parse(operatorCookie);
+    const { operator } = operatorInfo;
```
Comment (dannydaviesIW) [inline src/pages/service.tsx:89]: Nice bit of tidying up 👍

File: src/pages/service.tsx [hunk 2]
```diff
@@ -59,10 +66,18 @@ const Service = ({ operator, passengerType, services }: ServiceProps): ReactElem
     </Layout>
 );
 
-export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
-    deleteCookieOnServerSide(ctx, SERVICE_COOKIE);
-
+export const getServerSideProps = async (ctx: NextPageContext): Promise<{ props: ServiceProps }> => {
     const cookies = parseCookies(ctx);
+    const serviceCookie = cookies[SERVICE_COOKIE];
```
Comment (dannydaviesIW) [inline src/pages/service.tsx:71]: Do we want the cookie deleted server side still? We could extract the cookies and then use the `deleteCookieOnServerSide` method?

File: src/pages/singleDirection.tsx [hunk 3]
```diff
@@ -57,9 +64,17 @@ const SingleDirection = ({ operator, passengerType, lineName, service }: Directi
     );
 };
 
-export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
-    deleteCookieOnServerSide(ctx, JOURNEY_COOKIE);
+export const getServerSideProps = async (ctx: NextPageContext): Promise<{ props: DirectionProps }> => {
     const cookies = parseCookies(ctx);
+    const journeyCookie = cookies[JOURNEY_COOKIE];
```
Comment (dannydaviesIW) [inline src/pages/singleDirection.tsx:69]: Same here for the `deleteCookieOnServerSide` method

File: src/pages/service.tsx [hunk 4]
```diff
@@ -80,7 +96,7 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
         throw new Error(`No services found for NOC Code: ${operatorInfo.nocCode}`);
     }
 
-    return { props: { operator: operatorInfo.operator, passengerType, services } };
+    return { props: { operator, passengerType, services, error: errorInfo } };
```
Comment (dannydaviesIW) [inline src/pages/service.tsx:99]: Similar to the tidying up above, could we have `errorInfo` as just `error`?

File: src/pages/singleDirection.tsx [hunk 5]
```diff
@@ -97,7 +112,7 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
     );
 
     return {
-        props: { operator: operatorInfo.operator, passengerType, lineName, service },
+        props: { operator: operatorInfo.operator, passengerType, lineName, service, error: errorInfo },
```
Comment (dannydaviesIW) [inline src/pages/singleDirection.tsx:115]: Same as previous file 😄

File: src/pages/service.tsx [hunk 6]
```diff
@@ -1,24 +1,29 @@
...
+    error: ErrorInfo[];
 };
 
-const Service = ({ operator, passengerType, services }: ServiceProps): ReactElement => (
+const Service = ({ operator, passengerType, services, error }: ServiceProps): ReactElement => (
     <Layout title={title} description={description}>
         <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
             <form action="/api/service" method="post">
+                <ErrorSummary errors={error} />
                 <div className="govuk-form-group">
```
Comment (dannydaviesIW) [inline src/pages/service.tsx:27]: I think this line would need to be <div className={`govuk-form-group ${error.length > 0 ? 'govuk-form-group--error' : ''}`}> to get the red line alongside it

File: src/pages/singleDirection.tsx [hunk 7]
```diff
@@ -3,26 +3,31 @@ import { NextPageContext } from 'next';
...
 }
 
-const SingleDirection = ({ operator, passengerType, lineName, service }: DirectionProps): ReactElement => {
+const SingleDirection = ({ operator, passengerType, lineName, service, error }: DirectionProps): ReactElement => {
     return (
         <Layout title={title} description={description}>
             <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
                 <form action="/api/singleDirection" method="post">
+                    <ErrorSummary errors={error} />
                     <div className="govuk-form-group">
```
Comment (dannydaviesIW) [inline src/pages/singleDirection.tsx:31]: Same with this line being <div className={govuk-form-group ${error.length > 0 ? 'govuk-form-group--error' : ''}}> to get the red line alongside it

File: src/pages/service.tsx (hunk 2 above)
Comment (RobCatton) [reply src/pages/service.tsx:71]: I'm not sure what purpose it serves by deleting it - on some pages its needed for the logic (the way it is written anyway) but on this page I dont think it needs to be deleted. Can you think of a scenario we should?

File: src/pages/service.tsx (hunk 6 above)
Comment (RobCatton) [reply src/pages/service.tsx:27]: Added, thanks!

**Pull request merged**